server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java

```diff
@@ -304,22 +307,6 @@ public boolean plan(DefaultIssue issue, @Nullable ActionPlan actionPlan, IssueCh
     return false;
   }
 
-  public boolean setProject(DefaultIssue issue, String projectKey, IssueChangeContext context) {
-    if (!Objects.equal(projectKey, issue.projectKey())) {
-      issue.setProjectKey(projectKey);
-      issue.setUpdateDate(context.date());
-      issue.setChanged(true);
-      return true;
-    }
-    return false;
-  }
-
-  public boolean setPastProject(DefaultIssue issue, String previousKey, IssueChangeContext context) {
-    String currentProjectKey = issue.projectKey();
-    issue.setProjectKey(previousKey);
-    return setProject(issue, currentProjectKey, context);
-  }
-
   public boolean setTags(DefaultIssue issue, Collection<String> tags, IssueChangeContext context) {
     Set<String> newTags = Sets.newHashSet(Collections2.transform(
       Collections2.filter(tags, new Predicate<String>() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this a named "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2694)